lib/errors.js

```diff
@@ -285,6 +285,9 @@ internals.annotate = function (stripColorCodes) {
                         delete ref[replacement];
                     }
                     else {
+                        if (typeof ref === 'string') {
```

Comment:
@Marsup I fixed the issue with the `'_$miss$_forbiddenKey|1_$end$_'` TypeError.
You are talking about an issue with error annotation.
I added a failing test case for the missing `[1]`, but I don't know how to fix it yet.
Moving the `typeof ref === 'string'` check to the very 1st if of the loop doesn't help.